src/quo/components/profile/collectible_list_item/view.cljs

```diff
@@ -41,17 +43,9 @@
     :color-index     gradient-color-index}])
 
 (defn- card-details
-  [{:keys [status community? avatar-image-src collectible-name theme]}]
+  [{:keys [community? avatar-image-src collectible-name theme loading? error?]}]
   [rn/view {:style style/card-details-container}
-
-   (cond (= :cant-fetch status)
-         [text/text
-          {:size   :paragraph-1
-           :weight :medium
-           :style  {:color (colors/theme-colors colors/neutral-50 colors/neutral-40 theme)}}
-          (i18n/label :t/unknown)]
-
-         (= :loading status)
+   (cond @loading?
```

Comment:
hmm, better to pass de-ref value imo, unless it's really needed.Then you don't need to know the data type

Reply:
Seems atoms are being passed because they are being reset in this function.

@mmilad75 I'd suggest pass the values derefed and pass some callbacks performing the resets you are willing to do.

This reduces the area of errors, we've had many exceptions due to derefing on `nil`, forgetting to deref or derefing on an invalid type (that `IDeref` exception that you've probably seen).

Also this approach allow us to define the functions to mutate atoms only once, meaning sometimes we could avoid unnnecessary re-renders.